src/main/java/ui/UI.java

```diff
@@ -0,0 +1,186 @@
+package ui;
+
+import de.vandermeer.asciitable.AsciiTable;
+import de.vandermeer.asciitable.CWC_LongestLine;
+import de.vandermeer.skb.interfaces.transformers.textformat.TextAlignment;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.stream.Collectors;
+
+import exceptions.SEPEmptyException;
+import student.Student;
+import university.University;
+import university.UniversityRepository;
+
+/**
+ * Handles user interface FindMySEP application.
+ * Methods to print messages, retrieve user input, display ASCII table
+ * for student data.
+ */
+public class UI {
+    private static final int LINE_LENGTH = 80;
+    public static final String HORIZONTAL_LINE = "-".repeat(LINE_LENGTH);
+    private final Scanner scanner;
+
+    /**
+     * Constructor for UI class
+     */
+    public UI() {
+        this.scanner = new Scanner(System.in);
+    }
+
+    /**
+     * Prints the given item to the console, wrapped by horizontal lines above
+     * and below the specified item.
+     *
+     * @param item The item to be printed, which can be of any type.
+     *             If the item is null, it prints "null".
+     */
+    public <T> void printResponse(T item) {
+        String content = (item == null) ? "null" : item.toString();
+        System.out.println(HORIZONTAL_LINE + "\n" + content + "\n" + HORIZONTAL_LINE);
+    }
+
+    /**
+     * Retrieves a line of user input from the console.
+     *
+     * @return The line of input entered by the user, or an empty string
+     *         if an error occurs while reading input.
+     */
+    public String getUserInput() {
+        try {
+            return scanner.nextLine().trim();
+        } catch (Exception e) {
+            printResponse("Error reading input: " + e.getMessage());
+            return ""; // Return an empty string
+        }
+    }
+
+    /**
+     * Prints a greeting message to the console.
+     */
+    public void sayHi() {
+        printResponse(Messages.WELCOME);
+    }
+
+    /**
+     * Prints a farewell message to the console.
+     */
+    public void cleanupAndExit() {
+        printResponse("Adios, amigo!");
+        scanner.close();
+    }
+
+    /**
+     * Prints table displaying student information.
+     * The table includes student IDs, GPAs, and preference rankings.
+     *
+     * @param studentList ArrayList of Student objects to be printed
+     * @throws SEPEmptyException if the student list is null or empty.
+     */
+    public void printStudentList(ArrayList<Student> studentList) throws SEPEmptyException {
+        if (studentList == null || studentList.isEmpty()) {
+            throw SEPEmptyException.rejectEmptyStudentList();
+        }
+
+        AsciiTable at = new AsciiTable();
```

Comment:
You could consider give the variable a more meaningful name by giving a bit more description.